app/lib/features/chat_ng/widgets/events/member_update_event.dart

```diff
@@ -63,4 +34,87 @@ class MemberUpdateEvent extends StatelessWidget {
       ),
     );
   }
+
+  String getStateEventStr(
+    BuildContext context,
+    WidgetRef ref,
+    RoomEventItem item,
+  ) {
+    final lang = L10n.of(context);
+
+    final senderId = item.sender();
+    final eventType = item.eventType();
+    final msgType = item.msgType();
+    final firstName = ref
+        .watch(memberDisplayNameProvider((roomId: roomId, userId: senderId)))
+        .valueOrNull;
+    final msgContent = item.msgContent()?.body() ?? '';
+
+    return switch (eventType) {
+      'ProfileChange' => switch (msgType) {
+          'ChangedDisplayName' =>
+            '${lang.chatDisplayNameUpdate(firstName ?? senderId)} $msgContent',
+          'SetDisplayName' =>
+            '${lang.chatDisplayNameSet(firstName ?? senderId)}: $msgContent',
+          'RemoveDisplayName' =>
+            lang.chatDisplayNameUnset(firstName ?? senderId),
+          'ChangeProfileAvatar' =>
+            lang.chatUserAvatarChange(firstName ?? senderId),
+          _ => msgContent
+        },
+      _ => switch (msgType) {
+          'Joined' => isMe
+              ? lang.chatYouJoined
+              : firstName != null
+                  ? lang.chatJoinedDisplayName(firstName)
+                  : lang.chatJoinedUserId(senderId),
+          'Left' => isMe
+              ? lang.chatYouLeft
+              : firstName != null
+                  ? lang.chatUserLeft(firstName)
+                  : lang.chatUserLeft(senderId),
```

Comment:
```suggestion
              : lang.chatUserLeft(firstName ?? senderId),
```

Similar things can be applies to all the other use-cases.